test/e2e/volume-provisioner/framework/framework.go

```diff
@@ -315,3 +302,373 @@ func (f *Framework) newConfigurationProvider(cfg *core.Config) (common.Configura
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "oci-flexvolume-driver-worker",
+					Labels: map[string]string{"app": "oci-flexvolume-driver-worker"},
+				},
+				Spec: v1.PodSpec{
+					ServiceAccountName: "oci-flexvolume-driver",
+					Containers: []v1.Container{{
+						Name:    "oci-fexvolume-driver",
+						Image:   TestContext.Image,
+						Command: []string{"/bin/bash", "/usr/local/bin/install.sh"},
+						SecurityContext: &v1.SecurityContext{
+							Privileged: utilpointer.BoolPtr(true),
+						},
+						VolumeMounts: []v1.VolumeMount{{
+							Name:      "flexvolume-mount",
+							MountPath: "/flexmnt",
+						}},
+					}},
+					Volumes: []v1.Volume{{
+						Name: "flexvolume-mount",
+						VolumeSource: v1.VolumeSource{
+							HostPath: &v1.HostPathVolumeSource{
+								Path: "/usr/libexec/kubernetes/kubelet-plugins/volume/exec/",
+								Type: &hostPathDirectoryOrCreate,
+							},
+						},
+					}},
+				},
+			},
+		},
+	}
+
+	if err := createAndAwaitDaemonSet(client, masterDS); err != nil {
+		return errors.Wrap(err, "installing flexvolume driver master DaemonSet")
+	}
+
+	if err := createAndAwaitDaemonSet(client, workerDS); err != nil {
+		return errors.Wrap(err, "installing flexvolume driver worker DaemonSet")
+	}
+
+	return nil
+}
+
+func createAndAwaitDaemonSet(client clientset.Interface, desired *appsv1.DaemonSet) error {
+	actual, err := client.AppsV1().DaemonSets("kube-system").Create(desired)
+	if err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return errors.Wrapf(err, "failed to create %q DaemonSet", desired.Name)
+		}
+		Logf("Flexvolume Driver DaemonSet already exists. Updating.")
+		actual, err = client.AppsV1().DaemonSets("kube-system").Update(desired)
+		if err != nil {
+			return errors.Wrapf(err, "updating flexvolume driver DaemonSet %q", desired.Name)
+		}
+	} else {
+		Logf("Created DaemonSet %q in namespace %q", actual.Name, actual.Namespace)
+	}
+
+	return wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+		actual, err := client.AppsV1().DaemonSets(actual.Namespace).Get(actual.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, errors.Wrap(err, "waiting for DaemonSet to be ready")
+		}
+
+		if actual.Status.DesiredNumberScheduled != 0 && actual.Status.NumberReady == actual.Status.DesiredNumberScheduled {
+			return true, nil
+		}
+
+		Logf("%s DaemonSet not yet ready DaemonSet not yet ready (diesired=%d, ready=%d). Waiting...",
+			actual.Name, actual.Status.DesiredNumberScheduled, actual.Status.NumberReady)
+		return false, nil
+	})
+}
+
+// DeleteFlexvolumeDriver deletes both the master and worker flexvolume driver
+// DaemonSets.
+func DeleteFlexvolumeDriver(client clientset.Interface) error {
+	Logf("Deleteing oci-flexvolume-driver-master DaemonSet")
+
+	// TODO(apryde): We can probably use a label selector to delete both at
+	// once as this currently leaves dangling resources if the first delete
+	// fails.
```

Comment:
We could also make both API calls and check the errors at the end.

Reply:
Can we also make this requests in parallel? They're not co-dependent

Reply:
> We could also make both API calls and check the errors at the end.

Yeah, I was just being lazy - will fix.

> Can we also make this requests in parallel? They're not co-dependent

Could do but in reality not worth the effort as they only take a handful of seconds to complete sequentially vs multiple minute test run.